matchzoo/engine/pipeline.py

```diff
@@ -0,0 +1,133 @@
+"""Pipeline definition, consist of multiple :class:`ProcessorUnit`.
+
+Each model-wise preprocessor should employ a sequence of :class:`ProcessorUnit`
+and :class:`StatefulProcessorUnit` to handle input data.
+"""
+
+import typing
+
+from matchzoo import preprocessor
+
+
+class Pipeline(object):
```

Comment:
I think this is the case where using a class doesn't add any functionality but complexity. I can provide a function that take a list of processor units and process them sequentially, and achieve the same functionalities as your class by simply managing the list and calling the function. Like we discussion, we don't need a class that's a fixed part of the preprocessing process (like a parent class that everyone inherits), but helper utilities that helps.

Reply:
That pattern doesn't fit here.